pyoptsparse/pyOpt_utils.py

```diff
@@ -600,7 +605,7 @@ def try_import_compiled_module_from_path(module_name: str, path: Optional[str] =
     try:
         module = importlib.import_module(module_name)
     except ImportError as e:
-        if path is not None:
+        if raise_warning:
```

Comment:
What are the scenarios where we don't want to raise a warning when the import fails? Even if it's from `THIS_DIR` we should probably raise an error

Reply:
Right now on the main branch, running e.g. `example/hs015.py` will emit the following warnings for me:
```
.../pyoptsparse/pyoptsparse/pyIPOPT/pyIPOPT.py:28: UserWarning: pyipoptcore module could not be imported from .../repos/pyoptsparse/pyoptsparse/pyIPOPT.
  pyipoptcore = try_import_compiled_module_from_path("pyipoptcore", THIS_DIR)
.../pyoptsparse/pyoptsparse/pyNLPQLP/pyNLPQLP.py:20: UserWarning: nlpqlp module could not be imported from .../pyoptsparse/pyoptsparse/pyNLPQLP.
  nlpqlp = try_import_compiled_module_from_path("nlpqlp", THIS_DIR)
```

Without the changes in this PR, we always print warnings (at import time) if we cannot import these libraries. Given that they are optional and not installed by pyOptSparse, I don't think we will want this for end users. The PR changes this behaviour so that warnings are not emitted by default. Note that errors are raised at instantiation time if import fails and this PR does not change that behaviour.